src/main/java/tr/com/srdc/cda2fhir/transform/ResourceTransformerImpl.java

```diff
@@ -2216,7 +2227,8 @@ public EntryResult tMedicationActivity2MedicationStatement(MedicationActivity cd
 		}
 
 		MedicationStatement fhirMedSt = new MedicationStatement();
-		org.hl7.fhir.dstu3.model.Dosage fhirDosage = fhirMedSt.addDosage();
+		org.hl7.fhir.r4.model.Dosage fhirDosage = fhirMedSt.addDosage();
+		DosageDoseAndRateComponent fhirDosageAndRate = fhirDosage.addDoseAndRate();
 		result.addResource(fhirMedSt);
```

Comment:
I think this object might end up empty but I guess that is ok?

Reply:
How will it end up empty?

Reply:
If cdaMedicationActivity.getDoseQuantity() and cdaMedicationActivity.getRateQuantity() are null or nullFlavor.

Reply:
got it